### PR TITLE
Migrate configs from `adstax` to `adstax.sdk`

### PR DESCRIPTION
**Pull request by queimadus**
This PR migrates the configs in `adstax` to `adstax.sdk`.

It also removes the hardcoded config path for creating the `JsonFormat` for `SchedulerJob`s.

### CODE REVIEW

File: src/main/scala/eu/shiftforward/adstax/tracking/TrackingListener.scala [hunk 1]
```diff
@@ -98,8 +98,8 @@ object TrackingListenerActor {
   def props(
     processorProps: Props,
     numProcessors: Int = 1,
-    amqpConfig: config.RabbitMQ = LazyConfigFactory.load.getConfig("adstax.event-tracker.rabbitmq").to[config.RabbitMQ].get,
-    trackingConfig: config.RabbitMQ.Binding = LazyConfigFactory.load.getConfig("adstax.event-tracker.http-tracking").to[config.RabbitMQ.Binding].get) = {
+    amqpConfig: config.RabbitMQ = LazyConfigFactory.load.getConfig("adstax.sdk.event-tracker.rabbitmq").to[config.RabbitMQ].get,
+    trackingConfig: config.RabbitMQ.Binding = LazyConfigFactory.load.getConfig("adstax.sdk.event-tracker.rmq-bindsings.http-events").to[config.RabbitMQ.Binding].get) = {
```
Comment (ruippeixotog) [inline src/main/scala/eu/shiftforward/adstax/tracking/TrackingListener.scala]: Typo here. Also, can you have a more appropriate name to this parameter?

File: src/main/scala/eu/shiftforward/adstax/scheduler/api/rpc/SchedulerAmqpRpcClient.scala [hunk 2]
```diff
@@ -25,15 +27,19 @@ class SchedulerAmqpRpcClient(val amqp: AmqpClient, val rpcConfig: config.RpcClie
 
 object SchedulerAmqpRpcClient {
 
-  def apply(amqp: AmqpClient, rpcConfig: config.RpcClient): SchedulerAmqpRpcClient =
+  def apply(amqp: AmqpClient, rpcConfig: config.RpcClient, jobTypesConfig: Map[String, String]): SchedulerAmqpRpcClient = {
```
Comment (ruippeixotog) [inline src/main/scala/eu/shiftforward/adstax/scheduler/api/rpc/SchedulerAmqpRpcClient.scala]: Can you rename `jobTypesConfig` to just `jobTypes`?

File: src/main/scala/eu/shiftforward/adstax/scheduler/api/rpc/SchedulerAmqpRpcClient.scala [hunk 3]
```diff
@@ -25,15 +27,19 @@ class SchedulerAmqpRpcClient(val amqp: AmqpClient, val rpcConfig: config.RpcClie
...
+    implicit val schedulerJobJsonFormat: SchedulerJobJsonFormat = new SchedulerJob.SchedulerJobJsonFormat(jobTypesConfig)
     new SchedulerAmqpRpcClient(amqp, rpcConfig)
+  }
 
   def apply(
-    amqpConfig: config.RabbitMQ = LazyConfigFactory.load.getConfig("adstax.scheduler.rabbitmq").to[config.RabbitMQ].get,
-    rpcConfig: config.RpcClient = LazyConfigFactory.load.getConfig("adstax.scheduler.rpc").to[config.RpcClient].get)(
+    amqpConfig: config.RabbitMQ = LazyConfigFactory.load.getConfig("adstax.sdk.scheduler.rabbitmq").to[config.RabbitMQ].get,
+    rpcConfig: config.RpcClient = LazyConfigFactory.load.getConfig("adstax.sdk.scheduler.rpc").to[config.RpcClient].get,
+    jobTypesConfig: Map[String, String] = LazyConfigFactory.load.getConfig("adstax.sdk.scheduler.job-types").to[Map[String, String]].get)(
```
Comment (ruippeixotog) [inline src/main/scala/eu/shiftforward/adstax/scheduler/api/rpc/SchedulerAmqpRpcClient.scala]: Can you rename `jobTypesConfig` to just `jobTypes`?

File: src/main/scala/eu/shiftforward/adstax/scheduler/api/job/SchedulerJob.scala [hunk 4]
```diff
@@ -76,20 +74,18 @@ object SchedulerJob extends Logging {
       .optionalField[RepetitionConfig]("repeat")
   }
 
-  implicit object SchedulerJobJsonFormat extends RootJsonFormat[SchedulerJob] {
+  class SchedulerJobJsonFormat(jobTypeConfig: Map[String, String]) extends RootJsonFormat[SchedulerJob] {
```
Comment (ruippeixotog) [inline src/main/scala/eu/shiftforward/adstax/scheduler/api/job/SchedulerJob.scala]: Can you rename `jobTypeConfig` to just `jobTypes`?

File: src/main/scala/eu/shiftforward/adstax/scheduler/api/JobStatus.scala [hunk 5]
```diff
@@ -28,5 +29,5 @@ object JobStatus {
     override protected def fieldName(orig: String) = orig.toLowerCase
   }
 
-  implicit val jobStatusFormat: RootJsonFormat[JobStatus] = familyFormat
+  implicit def jobStatusFormat(implicit sjjf: SchedulerJobJsonFormat): RootJsonFormat[JobStatus] = familyFormat
```
Comment (ruippeixotog) [inline src/main/scala/eu/shiftforward/adstax/scheduler/api/JobStatus.scala]: Can't you use `RootJsonFormat[SchedulerJob]` everywhere you ask for a `SchedulerJobJsonFormat`?

File: src/main/scala/eu/shiftforward/adstax/tracking/TrackingListener.scala (hunk 1 above)
Comment (queimadus) [inline src/main/scala/eu/shiftforward/adstax/tracking/TrackingListener.scala]: Fixed in 5aaaff6

File: src/main/scala/eu/shiftforward/adstax/scheduler/api/rpc/SchedulerAmqpRpcClient.scala (hunk 2 above)
Comment (queimadus) [inline src/main/scala/eu/shiftforward/adstax/scheduler/api/rpc/SchedulerAmqpRpcClient.scala]: Fixed in 8c71b41.

File: src/main/scala/eu/shiftforward/adstax/scheduler/api/rpc/SchedulerAmqpRpcClient.scala (hunk 3 above)
Comment (queimadus) [inline src/main/scala/eu/shiftforward/adstax/scheduler/api/rpc/SchedulerAmqpRpcClient.scala]: Fixed in 8c71b41.

File: src/main/scala/eu/shiftforward/adstax/scheduler/api/job/SchedulerJob.scala (hunk 4 above)
Comment (queimadus) [inline src/main/scala/eu/shiftforward/adstax/scheduler/api/job/SchedulerJob.scala]: Fixed in 8c71b41.

File: src/main/scala/eu/shiftforward/adstax/scheduler/api/JobStatus.scala (hunk 5 above)
Comment (queimadus) [inline src/main/scala/eu/shiftforward/adstax/scheduler/api/JobStatus.scala]: Fixed in 4bf6357.

**Pull request merged**